src/app/home/problem-list/problem-list.component.ts

```diff
@@ -1,12 +1,24 @@
 import { Component, OnInit } from '@angular/core';
+import { ProblemListService } from './problem-list.service';
+import { Problem } from './problem.model';
 
 @Component({
   selector: 'app-problem-list',
   templateUrl: './problem-list.component.html',
   styleUrls: ['./problem-list.component.scss']
 })
 export class ProblemListComponent implements OnInit {
-  constructor() {}
+  problems: Problem[];
+  constructor(private problemListService: ProblemListService) {}
 
-  ngOnInit(): void {}
+  ngOnInit() {
+    this.problemListService.getProblems().subscribe((data) => {
+      this.problems = data.map((e) => {
+        return {
+          id: e.payload.doc.id,
+          ...(e.payload.doc.data() as Record<string, unknown>)
+        } as Problem;
+      });
+    });
```

Comment:
We will need to handle error. In case you want to do it later, add a `TODO` comment.  I will try to look in next sprint if we can add an interceptor for handling errors.

Reply:
Added TODO comment for handling errors
[Commit link](https://github.com/armory-uci/vital/pull/8/commits/066d6038cd7ea68ebd6bcc242bd6fed17dc374fb)